source/Calamari/Kubernetes/KubernetesContextScriptWrapper.cs

```diff
@@ -1,79 +1,717 @@
-            var k8sContextScriptFile = Path.Combine(workingDirectory, $"Octopus.{contextFile}");
-            var contextScript = embeddedResources.GetEmbeddedResourceText(Assembly.GetExecutingAssembly(), $"Calamari.Kubernetes.Scripts.{contextFile}");
-            fileSystem.OverwriteFile(k8sContextScriptFile, contextScript);
-            return k8sContextScriptFile;
+            public CommandResult Execute(string accountType)
+            {
+                var errorResult = new CommandResult(string.Empty, 1);
+
+                foreach (var proxyVariable in ProxyEnvironmentVariablesGenerator.GenerateProxyEnvironmentVariables())
+                {
+                    environmentVars[proxyVariable.Key] = proxyVariable.Value;
+                }
+
+                var kubeConfig = ConfigureCliExecution();
+
+                if (!TrySetKubectl())
+                {
+                    return errorResult;
+                }
+
+                var @namespace = variables.Get(SpecialVariables.Namespace);
+                if (string.IsNullOrEmpty(@namespace))
+                {
+                    log.Verbose("No namespace provided. Using default");
+                    @namespace = "default";
+                }
+
+                if (!TrySetupContext(kubeConfig, @namespace, accountType))
+                {
+                    return errorResult;
+                }
+
+                CreateNamespace(@namespace);
+
+                var outputKubeConfig = variables.GetFlag(SpecialVariables.OutputKubeConfig);
+                if (outputKubeConfig)
+                {
+                    ExecuteKubectlCommand("config", "view");
+                }
+
+                return new CommandResult(string.Empty, 0);
+            }
+
+            bool TrySetupContext(string kubeConfig, string @namespace, string accountType)
+            {
+                var clusterUrl = variables.Get(SpecialVariables.ClusterUrl);
+                var clientCert = variables.Get("Octopus.Action.Kubernetes.ClientCertificate");
+                var eksUseInstanceRole = variables.GetFlag("Octopus.Action.AwsAccount.UseInstanceRole");
+                var podServiceAccountTokenPath = variables.Get("Octopus.Action.Kubernetes.PodServiceAccountTokenPath");
+                var serverCertPath = variables.Get("Octopus.Action.Kubernetes.CertificateAuthorityPath");
+                var isUsingPodServiceAccount = false;
+                var skipTlsVerification = variables.GetFlag(SpecialVariables.SkipTlsVerification) ? "true" : "false";
+
+                if (accountType != "AzureServicePrincipal" && string.IsNullOrEmpty(clusterUrl))
+                {
+                    log.Error("Kubernetes cluster URL is missing");
+                    return false;
+                }
+
+                string podServiceAccountToken = null;
+                string serverCert = null;
+                if (string.IsNullOrEmpty(accountType) && string.IsNullOrEmpty(clientCert) && eksUseInstanceRole == false)
```

Comment:
This logic is going to get a lot more complex when we add gke with inherited service accounts, azure with managed service identity. It might be with extracting this if logic into a function.